py/kubeflow/testing/argo_client.py

```diff
@@ -44,23 +44,26 @@ def wait_for_workflows(client, namespace, names,
   crd_api = k8s_client.CustomObjectsApi(client)
   end_time = datetime.datetime.now() + timeout
   while True:
-    all_results = []
-
-    for n in names:
-      results = crd_api.get_namespaced_custom_object(
-          GROUP, VERSION, namespace, PLURAL, n)
-
-      all_results.append(results)
-      if status_callback:
-        status_callback(results)
-
-    done = True
-    for results in all_results:
-      if results["status"]["phase"] not in ["Failed", "Succeeded"]:
-        done = False
-
-    if done:
-      return all_results
+    try:
+      all_results = []
+
+      for n in names:
+        results = crd_api.get_namespaced_custom_object(
+            GROUP, VERSION, namespace, PLURAL, n)
+
+        all_results.append(results)
+        if status_callback:
+          status_callback(results)
+
+      done = True
+      for results in all_results:
+        if results["status"]["phase"] not in ["Failed", "Succeeded"]:
+          done = False
+
+      if done:
+        return all_results
+    except Exception as e: # pylint: disable=broad-except
```

Comment:
Why not use the retrying module
https://pypi.org/project/retrying/

So that we will give up after some number of retries as opposed to waiting forever.

I think we can just decorate wait_for_workflows with @retry

Reply:
Done